### PR TITLE
Closes #51

### LINKED ISSUE #51: Create a dataset loader for SCAI disease
From http://www.scai.fraunhofer.de/disease-ae-corpus.html

### PR DESCRIPTION
**Pull request by stephenbach**
- **Name:** SCAI Disease
- **Description:** https://www.scai.fraunhofer.de/en/business-research-areas/bioinformatics/downloads/corpus-for-disease-names-and-adverse-effects.html

### Checkbox

- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `biodatasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_BIGBIO_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `BigBioConfig` for the source schema and one for a bigbio schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_bigbio biodatasets/my_dataset/my_dataset.py`.
- [x] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

File: biodatasets/scai_disease/scai_disease.py [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+        urls = _URLS[_DATASETNAME]
+        data_dir = dl_manager.download(urls)
```
Comment (galtay) [inline biodatasets/scai_disease/scai_disease.py:137]: `filepath = dl_manager.download(urls)`

File: biodatasets/scai_disease/scai_disease.py [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "filepath": os.path.join(data_dir, "Disease-ae-corpus.iob"),
```
Comment (galtay) [inline biodatasets/scai_disease/scai_disease.py:143]: `"filepath": filepath`

File: biodatasets/scai_disease/scai_disease.py [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+        urls = _URLS[_DATASETNAME]
```
Comment (galtay) [inline biodatasets/scai_disease/scai_disease.py:136]: small thing, but for our future selves ... could we make `urls` singular? when you do `_URLS[_DATASETNAME]` you just get back a single string that is one url right?

File: biodatasets/scai_disease/scai_disease.py [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+
+_SUPPORTED_TASKS = [Tasks.NAMED_ENTITY_RECOGNITION]
+
+_SOURCE_VERSION = "1.0.0"
+
+_BIGBIO_VERSION = "1.0.0"
+
+
+class ScaiDiseaseDataset(datasets.GeneratorBasedBuilder):
+    """SCAI Disese is a dataset annotated in 2010 with mentions of diseases and
```
Comment (stephenbach) [inline biodatasets/scai_disease/scai_disease.py:69]:
```suggestion
    """SCAI Disease is a dataset annotated in 2010 with mentions of diseases and
```

File: biodatasets/scai_disease/scai_disease.py [hunk 5]
```diff
@@ -0,0 +1,260 @@
...
+  title     = {An Empirical Evaluation of Resources for the Identification of Diseases and Adverse Effects in Biomedical Literature},
+  booktitle = {LREC Workshop on Building and Evaluating Resources for Biomedical Text Mining},
+  year      = {2010},
+}
+"""
+
+_DATASETNAME = "[scai_disease]"
+
+_DESCRIPTION = """\
+SCAI Disese is a dataset annotated in 2010 with mentions of diseases and
```
Comment (stephenbach) [inline biodatasets/scai_disease/scai_disease.py:46]:
```suggestion
SCAI Disease is a dataset annotated in 2010 with mentions of diseases and
```

File: biodatasets/scai_disease/scai_disease.py [hunk 6]
```diff
@@ -0,0 +1,260 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+A dataset loader for the SCAI Disease dataset.
+
+SCAI Disese is a dataset annotated in 2010 with mentions of diseases and
```
Comment (stephenbach) [inline biodatasets/scai_disease/scai_disease.py:19]:
```suggestion
SCAI Disease is a dataset annotated in 2010 with mentions of diseases and
```

File: biodatasets/scai_disease/scai_disease.py [hunk 7]
```diff
@@ -0,0 +1,260 @@
...
+_CITATION = """\
+@inproceedings{gurulingappa:lrec-ws10,
+  author    = {Harsha Gurulingappa and Roman Klinger and Martin Hofmann-Apitius and Juliane Fluck},
+  title     = {An Empirical Evaluation of Resources for the Identification of Diseases and Adverse Effects in Biomedical Literature},
+  booktitle = {LREC Workshop on Building and Evaluating Resources for Biomedical Text Mining},
+  year      = {2010},
+}
+"""
+
+_DATASETNAME = "[scai_disease]"
```
Comment (stephenbach) [inline biodatasets/scai_disease/scai_disease.py:43]:
```suggestion
_DATASETNAME = "scai_disease"
```

**Review by galtay — APPROVED** [review]
nice work!

**Pull request merged**